### PR TITLE
Polygon Reward Claim Integration

### PR DESCRIPTION
**Pull request by viraj124**

### CODE REVIEW

**Review by fodisi — COMMENTED** [review]
@viraj124 
Can't we use inheritance to implement the Polygon Reward Claim? we're duplicating a lot on unnecessary code IMO. Let me know your thoughts.

Thank you.

**Comment by viraj124** [comment]
> @viraj124
> Can't we use inheritance to implement the Polygon Reward Claim? we're duplicating a lot on unnecessary code IMO. Let me know your thoughts.
> 
> Thank you.

Hey @fodisi 
1. If we inherit the contract will have to set up it's constructor parameters and this is an save polygon contract.
2. GoodGhosting.sol refers to the Ethereum network only and the aave claim Contract isn't deployed on Ethereum hence the different files 1 for Ethereum and others for polygon

**Comment by viraj124** [comment]
@fodisi maybe I misunderstood do you mean our polygon Contract inherits goodghosting.sol and override the redeem function sorry I misunderstood

**Comment by fodisi** [comment]
> @fodisi maybe I misunderstood do you mean our polygon Contract inherits goodghosting.sol and override the redeem function sorry I misunderstood

Yeah, pretty much. Adds the additional constructor parameter for the claim rewards and overrides the redeem function.
What do you think?

**Comment by viraj124** [comment]
> > @fodisi maybe I misunderstood do you mean our polygon Contract inherits goodghosting.sol and override the redeem function sorry I misunderstood
> 
> Yeah, pretty much. Adds the additional constructor parameter for the claim rewards and overrides the redeem function.
> What do you think?

Yeah will do that

**Comment by viraj124** [comment]
@fodisi pushed please check

File: contracts/GoodGhosting_Polygon.sol [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+import "./quickswap/IRouter.sol";
+import "./GoodGhostingWhitelisted.sol";
+import "./GoodGhosting.sol";
+
+/**
+ * Play the save game.
+ *
+ */
+
+contract GoodGhosting_Polygon is Ownable, Pausable, GoodGhosting {
```
Comment (fodisi) [inline contracts/GoodGhosting_Polygon.sol:22]: GoodGhosting is already pausable and ownable, so no need to define them again.

```suggestion
contract GoodGhosting_Polygon is GoodGhosting {
```

File: contracts/GoodGhosting_Polygon.sol [hunk 2]
```diff
@@ -0,0 +1,144 @@
+// SPDX-License-Identifier: UNLICENSED
+
+pragma solidity 0.6.11;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/utils/Pausable.sol";
```
Comment (fodisi) [inline contracts/GoodGhosting_Polygon.sol:6]: See comment below, but I believe we can remove these imports

```suggestion
```

File: contracts/GoodGhosting_Polygon.sol [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+        uint256 _segmentPayment,
+        uint256 _earlyWithdrawalFee,
+        uint256 _customFee,
+        address _dataProvider,
+        address _incentiveController,
+        IRouter _router,
+        IERC20 _matic,
+        IERC20 _quick,
+        bytes32 merkleRoot_
+    )
```
Comment (fodisi) [inline contracts/GoodGhosting_Polygon.sol:53]: I'd suggest you pass the new parameters at the end, to make it more similar to the GoodGhosting contract (for similarity and maintainability reasons, only).

```suggestion
    constructor(
        IERC20 _inboundCurrency,
        ILendingPoolAddressesProvider _lendingPoolAddressProvider,
        uint256 _segmentCount,
        uint256 _segmentLength,
        uint256 _segmentPayment,
        uint256 _earlyWithdrawalFee,
        uint256 _customFee,
        address _dataProvider,
        bytes32 merkleRoot_,
        address _incentiveController,
        IRouter _router,
        IERC20 _matic,
        IERC20 _quick
    )
```

**Review by fodisi — COMMENTED** [review]
@viraj124 Some minor comments. Looking good.

File: contracts/GoodGhosting_Polygon.sol (hunk 1 above)
Comment (viraj124) [reply contracts/GoodGhosting_Polygon.sol:22]: my bad missed this

Comment (fodisi) [reply contracts/GoodGhosting_Polygon.sol]: no worries ;)

**Comment by viraj124** [comment]
@fodisi all changes done check when you get time
Thanks :)

**Review by fodisi — APPROVED** [review]

**Pull request merged**